### DISCUSSION TITLE
Virtual switch dont return all interfaces

### DISCUSSION THREAD

**Discussion by rsp1991** [netbox-community/netbox#16186] [Help Wanted!]
### Deployment Type

NetBox Enterprise

### NetBox Version

v3.7.3

### Python Version

3.8

### Steps to Reproduce

1 - Get the Master device of the Virtual Switch with switch switches
2 - Go to https://<Netbox IP>/api/dcim/interfaces/?device=<device name>
3 - The output show only interfaces from Master switch.

### Expected Behavior

The expected is the Master return all interface

### Observed Behavior

Return all interfaces

**Comment by arthanson** [comment]
Thank you for opening a bug report. Unfortunately, the information you have provided is not sufficient for someone else to attempt to reproduce the reported behavior. Remember, each bug report must include detailed steps that someone else can follow on a clean, empty NetBox installation to reproduce the exact problem you're experiencing. These instructions should include the creation of any involved objects, any configuration changes, and complete accounting of the actions being taken. Also be sure that your report does not reference data on the public NetBox demo, as that is subject to change at any time by an outside party and cannot be relied upon for bug reports.

**Comment by rsp1991** [comment]
Hello @arthanson 

Sorry, but I don't have too much ideia to explain, I will try again:

I configured one virtual switch with two cisco switch with 48 ports,

SW1 48 Ports - Master
SW2 48 Ports

When I user the device.interfaces.all() I get just interfaces from SW1, I checked via URL:

https://<netbox_IP>api/dcim/interfaces/?device=<device>

And I confirmed that API answer the requisition with just SW1.

All interfaces in SW1 start with 1 and in the SW2 start with 2 so I don't use same name.

**Comment by jeffgdotorg** [comment]
It seems you're looking for advice on how to accomplish a task, rather than reporting a bug in the NetBox code that needs to be fixed. I'm therefore converting this issue to a discussion so that others in the community can help you accomplish your goal.
